### PR TITLE
Replace modulo operator in `Data.__getitem__()` with list

### PR DESCRIPTION
**Pull request by swapneelap**

### CODE REVIEW

File: micromagneticdata/data.py [hunk 1]
```diff
@@ -169,7 +169,11 @@ def __getitem__(self, item):
         Drive(...)
 
         """
-        return md.Drive(name=self.name, number=item % self.n, dirname=self.dirname)
+        return md.Drive(
+            name=self.name,
+            number=[i for i in range(self.n)][item],
```
Comment (lang-m) [inline micromagneticdata/data.py:174]: `number=range(self.n)[item]`

Comment (swapneelap) [reply micromagneticdata/data.py:174]: Good point! I was under the impression that `range()` is genexp.

Comment (swapneelap) [reply micromagneticdata/data.py]: I see.

**Review by lang-m — APPROVED** [review]
LGTM

**Pull request merged**